### PR TITLE
Seasonality modularization

### PR DESCRIPTION
**Pull request by karl-richter**
## :microscope: Background

- Following the example of trend, the seasonality component is modularized.

## :crystal_ball: Key changes

- Created a new Seasonality abstract and FourierSeasonality class.
- Find the new class diagram [here](https://github.com/ourownstory/neural_prophet/tree/feature/modular_seasonality/neuralprophet/components).

## :clipboard: Review Checklist
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, added docstrings and data types to function definitions.
- [x] I have added pytests to check whether my feature / fix works.

Please make sure to follow our best practices in the [Contributing guidelines](https://github.com/ourownstory/neural_prophet/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alfonsogarciadecorral** [comment]
Hey @karl-richter 
I'd suggest to follow the trend structure and creating Local/Global subclasses from `FourierSeasonality`.
It might not be that useful now, but it might be once we add Glocal.

**Comment by karl-richter** [comment]
> 

Good call @alfonsogarciadecorral , I adressed your comment and create a Global and Local class of Seasonality.

File: neuralprophet/components/seasonality/fourier.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        self.season_dims = utils.config_seasonality_to_model_dims(self.config_seasonality)
+        if self.season_dims is not None:
+            # Seasonality parameters for global or local modelling
+            self.season_params = nn.ParameterDict(
+                {
+                    # dimensions - [no. of quantiles, num_seasonalities_modelled, no. of fourier terms for each seasonality]
+                    name: init_parameter(dims=[len(self.quantiles)] + [self.num_seasonalities_modelled] + [dim])
+                    for name, dim in self.season_dims.items()
+                }
+            )
```
Comment (ourownstory) [inline neuralprophet/components/seasonality/fourier.py:30]: In a future PR, we may want to have a seasonality object per period instantiated - thus a class should not need know the `num_seasonalities_modelled`.

**Review by ourownstory — APPROVED** [review]
Great work!

**Pull request merged**